scipy/interpolate/_aaa.py

```diff
@@ -0,0 +1,404 @@
+# modification, are permitted provided that the following conditions are met:
+#     * Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in the
+#       documentation and/or other materials provided with the distribution.
+#     * Neither the name of the University of Oxford nor the names of its
+#       contributors may be used to endorse or promote products derived from
+#       this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import warnings
+
+import numpy as np
+import scipy
+
+
+__all__ = ["AAA"]
+
+
+class AAA:
+    r"""
+    AAA real or complex rational approximation.
+
+    As described in [1]_, the AAA algorithm is a greedy algorithm for approximation by
+    rational functions on a real or complex set of points. The rational approximation is
+    represented in a barycentric form from which the roots (zeros), poles, and residues
+    can be computed.
+
+    Parameters
+    ----------
+    z : 1D array_like
+        Values at which `f` is provided.
+    f : 1D array_like
+        Function values ``f(z)`` at `z`. Infinite and NaN value of `f` and corresponding
+        values of `z` will be discarded.
+    rtol : float, optional
+        Relative tolerance, defaults to ``eps**0.75``.
+    max_terms : int, optional
+        Maximum number of terms in the barycentric representation, defaults to 100.
+
+    Attributes
+    ----------
+    support_points : array
+        Support points of the approximation.
+    values : array
+        Value of the approximation at the `support_points`.
+    weights : array
+        Weights of the barycentric approximation.
+    errors : array
+        Error in the successive iterations of AAA.
+    
+    Methods
+    -------
+    poles()
+        Compute the poles of the rational approximation.
+    residues()
+        Compute the residues associated with the `poles` of the rational approximation.
+    roots()
+        Compute the roots(zeros) of the rational approximation.
+    
+
```

Comment:
```suggestion
```
Currently it renders as a table without links:
<img width="600" alt="image" src="https://github.com/user-attachments/assets/4edafc41-3cb8-428f-b48e-e455cea48ae2">

I think if you delete this, it will automatically be generated with links to the method documentation. It looks like that was intended because they have docstrings with more information.